src/uu/split/src/split.rs

```diff
@@ -645,14 +630,18 @@ fn get_input_size<R>(
     input: &String,
     reader: &mut R,
     buf: &mut Vec<u8>,
-    io_blksize: &Option<usize>,
+    io_blksize: &Option<u64>,
 ) -> std::io::Result<u64>
 where
     R: BufRead,
 {
     // Set read limit to io_blksize if specified
-    // Otherwise to OPT_IO_BLKSIZE_MAX
-    let read_limit = io_blksize.unwrap_or(OPT_IO_BLKSIZE_MAX) as u64;
+    let read_limit: u64 = if let Some(custom_blksize) = io_blksize {
+        *custom_blksize
```

Comment:
Should this be bounded by `sane_blksize::MAX` or should we check that it's non-zero?

Reply:
This value is checked for MAX at line 506 when its read from the command line argument. I'll add a check for zero.

Reply:
I see checks for `0` and `MAX` now, so I think it's good now, right?